pkg/streamingpromql/engine_test.go

```diff
@@ -273,6 +283,279 @@ func TestRangeVectorSelectors(t *testing.T) {
+			expr: "histogram_with_stale_marker[3m]",
+			ts:   baseT.Add(3 * time.Minute),
+			expected: &promql.Result{
+				Value: promql.Matrix{
+					{
+						Metric: labels.FromStrings("__name__", "histogram_with_stale_marker"),
+						Histograms: []promql.HPoint{
+							{
+								T: timestamp.FromTime(baseT),
+								H: &histogram.FloatHistogram{
+									Sum:   1,
+									Count: 1,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 2,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 0,
+									},
+								},
+							},
+							{
+								T: timestamp.FromTime(baseT.Add(time.Minute)),
+								H: &histogram.FloatHistogram{
+									Sum:   2,
+									Count: 2,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 2,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 1,
+									},
+								},
+							},
+							{
+								T: timestamp.FromTime(baseT.Add(3 * time.Minute)),
+								H: &histogram.FloatHistogram{
+									Sum:   4,
+									Count: 4,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 4,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 1, 1, 1,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"mixed series with histograms and floats": {
+			expr: "mixed_metric[4m]",
+			ts:   baseT.Add(4 * time.Minute),
+			expected: &promql.Result{
+				Value: promql.Matrix{
+					{
+						Metric: labels.FromStrings("__name__", "mixed_metric"),
+						Floats: []promql.FPoint{
+							{
+								T: timestamp.FromTime(baseT.Add(time.Minute)),
+								F: 1,
+							},
+							{
+								T: timestamp.FromTime(baseT.Add(2 * time.Minute)),
+								F: 2,
+							},
+						},
+						Histograms: []promql.HPoint{
+							{
+								T: timestamp.FromTime(baseT.Add(3 * time.Minute)),
+								H: &histogram.FloatHistogram{
+									Sum:   3,
+									Count: 3,
+									PositiveSpans: []histogram.Span{
+										{
+											Offset: 0,
+											Length: 3,
+										},
+									},
+									PositiveBuckets: []float64{
+										1, 2, 1,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"mixed series with a float following a histogram causing lookback": {
+			// This is unexpected, but consistent behavior between the engines
+			// See: https://github.com/prometheus/prometheus/issues/14172
+			expr: "mixed_metric_histogram_first[2m]",
+			ts:   baseT.Add(2 * time.Minute),
+			expected: &promql.Result{
+				Value: promql.Matrix{
+					{
+						Metric: labels.FromStrings("__name__", "mixed_metric_histogram_first"),
+						Floats: []promql.FPoint{
+							{
+								T: timestamp.FromTime(baseT.Add(time.Minute)),
+								F: 1,
+							},
+						},
+					},
+				},
+			},
+		},
+		"mixed series with histogram following a float": {
```

Comment:
[nit] Same as above - let's have the words in the same order as the points:

```suggestion
		"mixed series with a float then a histogram": {
```